subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/continuous/ContinuousBuildTriggerHandler.java

```diff
@@ -24,23 +24,22 @@
 import java.util.concurrent.TimeUnit;
 
 public class ContinuousBuildTriggerHandler {
-    public static final String QUIET_PERIOD_SYSPROP = "org.gradle.internal.filewatch.quietperiod";
-
     private final BuildCancellationToken cancellationToken;
     private final ContinuousExecutionGate continuousExecutionGate;
     private final CountDownLatch changeOrCancellationArrived = new CountDownLatch(1);
     private final CountDownLatch cancellationArrived = new CountDownLatch(1);
-    private final long quietPeriod;
+    private final int quietPeriod;
```

Comment:
We should use `Duration` here as well.

Reply:
I doesn't make sense to use Duration as the field here, at least as long as we don't convert all the other logic to use some time classes as well. I'd leave this as it is. I changed the constructor parameter to be a `Duration`.

Reply:
I think it's strongly preferable to state units when they are relevant, especially when it comes to time. I see two options to resolve this: either call the field `quitePeriodInMillis`, or declare it as a `Duration`. I think the latter is superior to the former.

Reply:
I converted `ContinuousBuildTriggerHandler` to use `Duration` and `Instant`. I think this is nicer. The only downside I see there is that I need to create my own `Instant.now()` to use the monotonic `System.nanoTime()`, and that instant doesn't have a real connection to the wall time clock, though I think that is fine. WDYT?

Reply:
That was surprising at first, but as someone who messed up writing this many times before:

```java
                long remainingQuietPeriod = quietPeriod - (now - lastChangeAt);
                if (remainingQuietPeriod <= 0) {
```

...I think switching to `Instant` makes sense, too.